### PR TITLE
topic (monitoring): [gateway & ilb] collect diagnostic settings

### PR DESCRIPTION
**Pull request by ferantivero**

### CODE REVIEW

**Review by josevarela — APPROVED** [review]
Looks great!

File: infra-as-code/bicep/gateway.bicep [hunk 1]
```diff
@@ -285,5 +294,26 @@ resource appGateway 'Microsoft.Network/applicationGateways@2021-05-01' = {
   ]
 }
 
+// App Gateway diagnostics
+resource appGatewayDiagSettings 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = {
+  name: '${appGateway.name}-diagnosticSettings'
```
Comment (ckittel) [inline infra-as-code/bicep/gateway.bicep]:
```suggestion
  name: 'default'
```

File: infra-as-code/bicep/internalloadbalancer.bicep [hunk 2]
```diff
@@ -101,5 +110,26 @@ resource internalLoadBalancer 'Microsoft.Network/loadBalancers@2021-05-01' = {
   dependsOn: []
 }
 
+// Internal Load Balancer diagnostics
+resource internalLoadBalancerDiagSettings 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = {
+  name: '${internalLoadBalancer.name}-diagnosticSettings'
```
Comment (ckittel) [inline infra-as-code/bicep/internalloadbalancer.bicep]:
```suggestion
  name: 'default'
```

File: infra-as-code/bicep/gateway.bicep [hunk 3]
```diff
@@ -71,6 +74,12 @@ resource vnet 'Microsoft.Network/virtualNetworks@2022-11-01' existing =  {
 
 }
 
+// Log Analytics Workspace
+resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2021-12-01-preview' existing = {
+  scope: resourceGroup()
```
Comment (ckittel) [inline infra-as-code/bicep/gateway.bicep]:
```suggestion
```

File: infra-as-code/bicep/internalloadbalancer.bicep [hunk 4]
```diff
@@ -28,6 +31,12 @@ var ilbName = 'ilb-${baseName}'
 
 /*** EXISTING RESOURCES ***/
 
+// Log Analytics Workspace
+resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2021-12-01-preview' existing = {
+  scope: resourceGroup()
```
Comment (ckittel) [inline infra-as-code/bicep/internalloadbalancer.bicep]:
```suggestion
```

**Review by ckittel — CHANGES_REQUESTED** [review]
Consistency recommendations to align with the existing items around it.

**Review by ckittel — APPROVED** [review]

**Pull request merged**